jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1303,11 +1315,24 @@ public Field getField(String name) throws NoSuchFieldException, SecurityExceptio
  * @return		Class<?>[]
  *					the interfaces the receiver claims to implement.
  */
-public Class<?>[] getInterfaces()
-{
-	return J9VMInternals.getInterfaces(this);
+public Class<?>[] getInterfaces(){
+	if (cachedInterfaces == null){
+		if (cachedInterfacesOffset == -1) {
+			Field cachedInterfacesField;
+			try {
+				cachedInterfacesField = Class.class.getDeclaredField("cachedInterfaces"); //$NON-NLS-1$
+				cachedInterfacesOffset = getUnsafe().objectFieldOffset(cachedInterfacesField);
+			} catch (NoSuchFieldException e) {
+				throw newInternalError(e);
+			}
+		}
+		getUnsafe().putOrderedObject(this, cachedInterfacesOffset, J9VMInternals.getInterfaces(this));
```

Comment:
Note the unsafe API is renamed to putObjectRelease() in Java 9 and you need to use the Java preprocessor to generate the correct code depending on the Java version. See the other examples of calling putOrderedObject() in this file.

Reply:
Can this line be moved within the block `if (cachedInterfacesOffset == -1) {` to avoid setting `null` to `cachedInterfaces` for each call to `getInterfaces()` in case there is no interfaces specified in the receiver class?

Reply:
no interfaces returns an empty array and not null.